chrome-extension/content.js

```diff
@@ -101,23 +97,50 @@ async function createSidebar(content) {
   commentTextarea.setAttribute('placeholder', 'Note to self');
   form.appendChild(commentTextarea);
 
-  // Create select dropdown with options fetched from categories db
-  const dropdownLabel = customCreateElement('label', ['label', 'mt-2'], 'Add Clipping to a Category?');
-  dropdownLabel.htmlFor = 'categoryDropdown';
-  form.appendChild(dropdownLabel);
-  
   sidebar.appendChild(form);
 
-  const categoryDropdown = await getCategoryDropdown(currentUserId);
-  categoryDropdown.id = 'categoryDropdown';
-  form.appendChild(categoryDropdown);
+  const userEmailNote = customCreateElement('div', []);
+  let mongoId;
+
+  // Get logged-in user info
+  chrome.extension.sendMessage({}, async (userInfo) => {
+    const email = userInfo.email;
+    const googleId = userInfo.id;
+
+    if (googleId) {
+      try {
+        const response = await fetch(serverUrl + '/user/getUserId/' + googleId);
+        mongoId = await response.json();
+
+        // Create select dropdown with options fetched from categories db
+        const dropdownLabel = customCreateElement('label', ['label', 'mt-2'], 'Add Clipping to a Category?');
+        dropdownLabel.htmlFor = 'categoryDropdown';
+        form.appendChild(dropdownLabel);
+
+        const categoryDropdown = await getCategoryDropdown(mongoId);
```

Comment:
Is there any way we can restructure it so that all the form elements are drawn to the page and then just the attributes are applied after? It is a bit odd to see parts of the UI pop in. You can use a loading styling to indicate that affected form elements are not ready for use.

To be clear, this doesn't have to come in this PR, but it would be nice

Reply:
okay good suggestion. I will look into loading styling, but i think this will all come in a styling pr.